Meta/gn/build/write_cmake_config.py

```diff
@@ -80,9 +80,9 @@ def repl(m):
             in_line = '#define %s %d\n' % (var, 1 if values[var] else 0)
             unused_values.discard(var)
         elif in_line.startswith('#cmakedefine '):
-            _, var = in_line.split(None, 1)
+            _, var = in_line.rstrip().split(None, 1)
```

Comment:
Can you paste the input line that isn't processed directly? The comment in line 28 is consistent with the LHS of the code afaict, and this script has been used as-is for a long time in LLVM.

(It's still possible it's buggy of course, but it doesn't feel super likely to me.)

Reply:
https://github.com/LadybirdBrowser/ladybird/blob/master/AK/Backtrace.h.in#L9-L13

`Backtrace_FOUND=` is being passed, and the lack of the value is what was not being handled properly. The `rstrip()` change can be dropped of course.

Reply:
I did a bit more testing of this with the current state of master (without this patch)

The codegen action for that file has two branches:
```
  if (current_os == "win") {
    values = [
      "Backtrace_FOUND=",
      "Backtrace_HEADER=",
    ]
  } else {
    values = [
      "Backtrace_FOUND=1",
      "Backtrace_HEADER=execinfo.h",
    ]
  }
  ```
  
If we take the second branch, as on linux, the file looks correct:  
```
$ cat out/gen/AK/Backtrace.h 
/*
 * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */

#pragma once

#define Backtrace_FOUND
#if defined(Backtrace_FOUND)
#    define AK_HAS_BACKTRACE_HEADER
#    undef Backtrace_FOUND
#endif

#if defined(AK_HAS_BACKTRACE_HEADER)
#    include <execinfo.h>
#endif
```

If we take the first branch, it also does the "right thing", though the un-taken preprocessor branch looks a bit awkward. I think we would get the same behavior if CMake generated the thing.

```
$ cat out/gen/AK/Backtrace.h 
/*
 * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */

#pragma once

/* #undef Backtrace_FOUND */
#if defined(Backtrace_FOUND)
#    define AK_HAS_BACKTRACE_HEADER
#    undef Backtrace_FOUND
#endif

#if defined(AK_HAS_BACKTRACE_HEADER)
#    include <>
#endif
```

Is this perhaps a python version thing? I'm using 3.10.12 on Ubuntu.

Reply:
I believe that I was using 3.9.10 (yes, it is old, but Swift requires Python 3.9).

Reply:
Works for me with 3.9.6:

```
 % python3 --version
Python 3.9.6

% python3 Meta/gn/build/write_cmake_config.py -o tmp.txt  AK/Backtrace.h.in Backtrace_FOUND=1 Backtrace_HEADER= && cat tmp.txt
/*
 * Copyright (c) 2024, Andrew Kaster <akaster@serenityos.org>.
 *
 * SPDX-License-Identifier: BSD-2-Clause
 */

#pragma once

#define Backtrace_FOUND
#if defined(Backtrace_FOUND)
#    define AK_HAS_BACKTRACE_HEADER
#    undef Backtrace_FOUND
#endif

#if defined(AK_HAS_BACKTRACE_HEADER)
#    include <>
#endif
```

What does that command print for you?

Reply:
I have to admit I don't understand _why_ it works. The code looks like it shouldn't work.

…aha! I think 050ffddff3111f681a629eca98b518ea21421d09 broke this. Before that, the failing split went down the `except _:` block and got `var` from there.

Reply:
Does it work for you if you locally revert the change from 050ffddff3111f681a629eca98b518ea21421d09 to this script?

Reply:
whoops. I added that change to appease the CI linter. It didn't like `except _:` or `except:`. Looks like a suppression would have been a better idea

Reply:
According to the docs, a falling split() throws ValueError, so maybe that change is a no-op actually.

But then I don't understand why this PR here is needed. Things should work fine: the split throws, we go in the except block, it handles the "no 3rd token" case, all is well.

Reply:
Ah, okay - that explains a lot. I had fixed this issue this way rather than the change to the script that was made in 050ffdd. So, I suppose the question is, which change to keep?

Reply:
050ffdd is a no-op as far as I currently understand, so I'd go with that.

This here looks like it'd break the case where there _is_ a third token on the line (?)